manifests/zone.pp

```diff
@@ -175,12 +189,46 @@
   $also_notify = [],
   $ensure = present,
   $data_dir = $::dns::server::params::data_dir,
+  $forwarders = undef,
+# DEPRECATED, to be removed in the next major release
+  $allow_forwarder = [],
 ) {
 
   $cfg_dir = $dns::server::params::cfg_dir
 
   validate_array($allow_transfer)
+
   validate_array($allow_forwarder)
+  # deprecation notice for allow_forwarder
+  if size($allow_forwarder) > 0 {
+    warning('dns::zone parameter `allow_forwarder` deprecated in favor of `forwarders`')
+    notify { 'dns::zone parameter `allow_forwarder` deprecated in favor of `forwarders`': }
+  }
+
+  # assign $zone_forwarders to the list of forwarders to define for the
+  # zone.  an empty list means *no forwarders*.  set $zone_forwarders to
+  # undef to not define the forwarders list at all (and thereby default
+  # to the forwarders list defined in the global options).
+
+  if $forwarders != undef {
+    if is_bool($forwarders) {
+      if $forwarders {
+        $zone_forwarders = undef
+      } else {
+        $zone_forwarders = []
+      }
+    } else {
+      validate_array($forwarders)
+      $zone_forwarders = $forwarders
+    }
+  } else {
```

Comment:
Not sure about the style on this module, but this would read cleaner for me if lines 224-225 were replaced with `elsif`.